### ISSUE TITLE
Fetching entries from `client.entries` or `sync.each_item` yields different Entry objects

### ISSUE THREAD

**Issue by barraq** [contentful/contentful.rb#160]
Entries, i.e. resources of type `Contentful::Entry`, can be fetched in different ways (see below). However, depending on the method used to fetch them, the `Contentful::Entry` object returned are not identical: at least they behave differently.

Lets `E` be an entry with `entry.id = A96usFSlY4G0W4kwAqswk`. The two methods below behave differently:

**(1) Using `client.entries`**

```
entries = client.entries()
entries.each_entry do |entry|
    if entry.id === "A96usFSlY4G0W4kwAqswk"
        puts entry.fields
        puts entry.raw
    end
end
```

With this method, `entry.fields` will return the fields without the `locale` if there is only one locale. So for instance, if the entry has the ContentType `Post` that has a field `title` then `entry.fields` will return something like:

```
{:title=>"Seven Tips From Ernest Hemingway on How to Write Fiction", ... }
```

where `entry.raw` will return something like:

```
{"sys"=>{..., "id"=>"", "type"=>"Entry", "locale"=>"en"}, "fields"=>{"title"=>"Seven Tips From Ernest Hemingway on How to Write Fiction", ... }}
```

**(2) Using `sync.each_item`:**

```
sync = client.sync(initial: true)
sync.each_item do |resource|
     if resource.id === "A96usFSlY4G0W4kwAqswk"
        puts resource.fields
        puts resource.raw
    end
end
```

With this method, `entry.fields` will return:

```
{}
```

where `entry.raw` will return something like:

```
{"sys"=>{..., id"=>"", "type"=>"Entry" }, "fields"=>{"title"=>{"en"=>"Seven Tips From Ernest Hemingway on How to Write Fiction"}}}
```

so basically `entry.raw` differs between the two methods. In the last method there is no `locale` returned. Instead each `field` is provided with each locale.

---

**Question:** why would the same entry be represented differently in the first and the second method? Is that voluntary?

Please let me know if the explanation is not clear enough.

**Comment by dlitvakb** [comment]
Hey @barraq,

Sorry for the delay in answering this question,

The Sync API returns the same resources as if you would do a request to the Entries endpoint with the `locale: '*'` parameter. That's the difference you're noticing.

Cheers
